src/fb.c

```diff
@@ -1,36 +1,39 @@
 /* SPDX-License-Identifier: MIT */
 
+#include <assert.h>
+
 #include "fb.h"
+#include "string.h"
 #include "utils.h"
 #include "xnuboot.h"
 
-u32 *fb;
-int fb_s, fb_w, fb_h, fb_d;
-int scale = 1;
+#define CONSOLE_MAX_ROWS 60
+#define CONSOLE_MAX_COLS 80
```

Comment:
These need to be calculated based on the width and height of the screen and the font size, otherwise your writes will go out of bounds (and things won't display on the screen properly).

If it's a Mac Mini with a 800x600 monitor connected you're going to run out of bounds.